master/internal/db/postgres_experiments.go

```diff
@@ -152,3 +155,100 @@ GROUP BY batches_processed;`, &rows, experimentID, metricName, startTime)
 
 	return batches, endTime, nil
 }
+
+type snapshotWrapper struct {
+	TrialID int32     `db:"trial_id"`
+	Hparams []byte    `db:"hparams"`
```

Comment:
question: hparams need to come back with all over these? if we're getting snapshots for multiple metrics, may make sense to be a separate endpoint? i don't have a whole lot of context on hp viz, i just know it's going to be the largest part of all of this.

Reply:
Yeah this is for a plot that does only show 1 metric at a time. I could see us retrieving all the hparams once and then using them across all the plots. The downside is the places where we return lots of hparams tend to be limited to a subset of trials (unless we're using trialsSnapshot at a very low number of batches). @hkang1 What do you think about that?

Reply:
After thinking about this more, then yes we do want the Hparams with each row. We're only ever fetching 1 row from the steps or validation table for each trial in a query, so the hparams don't get repeated a bunch. I agree they'll be the biggest part of this, but this is for a parallel coordinates plot that does need all that data. There's another call coming (trialsSample) that also fetches HParams, but the 2 are rendered separately, and trialsSample limits the number of trials to a relatively small number (10, 25, etc) and fetches those hparams separately instead of in a big DB join.